src/main/scala/io/keen/client/scala/BatchWriterClient.scala

```diff
@@ -0,0 +1,208 @@
+package io.keen.client.scala
+
+import java.util.concurrent.{ Executors, ScheduledThreadPoolExecutor, TimeUnit }
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+import com.typesafe.config.{ Config, ConfigFactory }
+
+object BatchWriterClient {
+  val MinSendIntervalEvents: Long = 100
+  val MaxSendIntervalEvents: Long = 10000
+  val MinSendInterval: FiniteDuration = 60.seconds
+  val MaxSendInterval: FiniteDuration = 1.hour
```

Comment:
This is more of a design decision based on how we see people using our SDKs than anything else. The bounds aren't absolutely necessary, and one could work around them by implementing their own batching of course, which isn't a big deal, we just thought that implementing it this way would offer a clear delineation between sending events via `addEvent`/`addEvents` and `sendQueuedEvents` while offering some built-in overflow protection.

It might be better to replace the existing bounds with payload size based bounds at some point as we do have strict payload sizes that the API accepts/rejects (900k for single events, 10M for batches).